src/Microsoft.Extensions.DependencyModel/DependencyContextWriter.cs

```diff
@@ -115,7 +117,7 @@ private JObject WriteDependencies(IReadOnlyList<Dependency> dependencies)
         private JObject WriteLibraries(DependencyContext context)
         {
             var allLibraries =
-                context.RuntimeLibraries.Concat(context.CompileLibraries)
+                context.RuntimeLibraries.OfType<Library>().Concat(context.CompileLibraries)
```

Comment:
Again, use `Cast`. `Library` is a super-type so the cast is definitely going to succeeed